sled-agent/src/dump_setup.rs

```diff
@@ -504,14 +518,24 @@ impl DumpSetupWorker {
 
     async fn poll_file_archival(mut self) {
         info!(self.log, "DumpSetup poll loop started.");
+
+        // A oneshot which helps callers track when updates have propagated.
+        //
+        // This is particularly useful for disk expungement, when a caller
+        // wants to ensure that the dump device is no longer accessing an
+        // old device.
+        let mut last_update_complete_tx = None;
```

Comment:
the prefix `last_` is ever-so-slightly confusing to its purpose - there's no code path where it isn't `.take()`n later in the same iteration of the loop where it was `= Some()`'d. (of course, `current_` wouldn't be particularly clear either.. perhaps `update_and_archiving_complete_tx` or something of the sort?)

Reply:
How about `evaluation_and_archiving_complete_tx` ? The fact that it's optional, IMO, implies that it may or may not be set, but we do respect that contract if it is set within a loop.

Reply:
Done in https://github.com/oxidecomputer/omicron/pull/5965/commits/691bc85bbde60f7d365ed9dac4d2ba902b1333f9